packages/nodes-base/nodes/Schedule/ScheduleTrigger.node.ts

```diff
@@ -412,238 +410,110 @@ export class ScheduleTrigger implements INodeType {
+						};
+					}
 				}
-			}
 
-			if (interval[i].field === 'weeks') {
-				const hour = interval[i].triggerAtHour?.toString() as string;
-				const minute = fallbackToZero(interval[i].triggerAtMinute?.toString() as string);
-				const week = interval[i].weeksInterval as number;
-				const days = interval[i].triggerAtDay as IDataObject[];
-				const day = days.length === 0 ? '*' : days.join(',');
-				const cronTimes: string[] = [minute, hour, '*', '*', day];
-				const cronExpression = cronTimes.join(' ');
-				if (week === 1) {
-					const cronJob = new CronJob(
-						cronExpression,
-						async () => await executeTrigger({ activated: false } as IRecurencyRule),
-						undefined,
-						true,
-						timezone,
-					);
-					cronJobs.push(cronJob);
-				} else {
-					const cronJob = new CronJob(
-						cronExpression,
-						async () =>
-							await executeTrigger({
-								activated: true,
-								index: i,
-								intervalSize: week,
-								typeInterval: 'weeks',
-							} as IRecurencyRule),
-						undefined,
-						true,
-						timezone,
-					);
-					cronJobs.push(cronJob);
+				if (interval.field === 'weeks') {
+					const { weeksInterval } = interval;
+					if (weeksInterval !== 1) {
+						recurrence = {
+							activated: true,
+							index: i,
+							intervalSize: weeksInterval,
+							typeInterval: 'weeks',
+						};
+					}
 				}
-			}
 
-			if (interval[i].field === 'months') {
-				const month = interval[i].monthsInterval;
-				const day = interval[i].triggerAtDayOfMonth?.toString() as string;
-				const hour = interval[i].triggerAtHour?.toString() as string;
-				const minute = fallbackToZero(interval[i].triggerAtMinute?.toString() as string);
-				const cronTimes: string[] = [minute, hour, day, '*', '*'];
-				const cronExpression: string = cronTimes.join(' ');
-				if (month === 1) {
-					const cronJob = new CronJob(
-						cronExpression,
-						async () => await executeTrigger({ activated: false } as IRecurencyRule),
-						undefined,
-						true,
-						timezone,
-					);
-					cronJobs.push(cronJob);
-				} else {
-					const cronJob = new CronJob(
-						cronExpression,
-						async () =>
-							await executeTrigger({
-								activated: true,
-								index: i,
-								intervalSize: month,
-								typeInterval: 'months',
-							} as IRecurencyRule),
-						undefined,
-						true,
-						timezone,
-					);
-					cronJobs.push(cronJob);
+				if (interval.field === 'months') {
+					const { monthsInterval } = interval;
+					if (monthsInterval !== 1) {
+						recurrence = {
+							activated: true,
+							index: i,
+							intervalSize: monthsInterval,
+							typeInterval: 'months',
+						};
+					}
 				}
-			}
-		}
 
-		async function closeFunction() {
-			for (const cronJob of cronJobs) {
-				cronJob.stop();
-			}
-			for (const entry of intervalArr) {
-				clearInterval(entry);
+				try {
+					this.helpers.registerCron(cronExpression, async () => await executeTrigger(recurrence));
```

Comment:
In manual mode these are not cleaned up. But I guess we do want to validate that the Cron expression is valid?

Reply:
nvm you just fixed it 😄 but do we still want to validate the Cron expression for manual executions?

Reply:
I'll try adding that 👍🏽

Reply:
done